### PR TITLE
User and groups control panel

### PR DESCRIPTION
**Pull request by nileshgulia1**
#591 @csenger  already implemented users control panel i.e ability to add and search for users . 
<b>Implementations:</b>
- [x] Now we have the ability to add and display group name with users.
- [x] Ability to Delete Groups.
- [x] Ability to Search for Groups.

<b>Left overs:</b>
- [x] Ability to add members into  groups (possibly by clicking one of the created groups, there should be pop up for adding users (like in original Plone). Although I'm not clear how to group these members together because it don't see an endpoint in the schema, need discussion from members).

### CODE REVIEW

**Comment by sneridagh** [comment]
@nileshgulia1

We need it to match the look and feel in:
https://app.zeplin.io/project/5c76b54e0e8c876239917ea3/screen/5c76b59686d8b86292eecca7

Please, can you take care? The screens are mainly in mobile sizes, but easy to translate to big ones. For the tables, very important: make them responsive, so the overflow (if any) is hidden and scrollable horizontally. Like https://getbootstrap.com/docs/4.0/content/tables/#responsive-tables

BTW, also, all content area in Volto is enclosed into a Container (for responsiveness), so make sure it is.

Also, I've noticed that the modal is kind of broken (it's displaced to the bottom).

Thanks!

**Comment by sneridagh** [comment]
More, the password in the field is shown in clear text.
The search does not search on type.

**Comment by sneridagh** [comment]
@nileshgulia1 
The password field is visible.

Screenshot 2019-03-12 at 17 32 31

The user roles can be modified (take a look into Plone's users/groups panel).

In responsive, the actual table implementation is unsusable... don't you think the same? If the columns are stacked, how do you know which roles have the user?

File: theme/themes/pastanaga/modules/modal.overrides [hunk 1]
```diff
@@ -11,7 +11,8 @@
 }
 
 .ui.modal {
-  position: fixed;
-  top: 50%;
+  position: relative;
   left: 50%;
+  top: 50%;
```
Comment (nileshgulia1) [inline theme/themes/pastanaga/modules/modal.overrides]: I think this is related to #677

**Comment by nileshgulia1** [comment]
@sneridagh Will be there any performance/security issue if we somehow update roles of users in real time by clicking checkboxes.

Unlike in Plone we do have a save button after checking all boxes we can save the settings.

**Comment by tisto** [comment]
@nileshgulia1 I fixed the failing unit tests. They are passing now. Though, there are quite a few issues with the control panel:

Users and Groups (2)
Users and Groups (1)
Users and Groups

**Comment by nileshgulia1** [comment]
@tisto Thanks for fixing tests and reporting issues. I'll look into it further at next week.

**Comment by tisto** [comment]
@nileshgulia1 awesome! Let me know if there is anything I can do to help!

**Comment by nileshgulia1** [comment]
@tisto  I fixed the highlighted issues, Can you please have a look now?

**Comment by tisto** [comment]
@nileshgulia1 will do. thanks!

**Comment by tisto** [comment]
@nileshgulia1 I gave it another try. Here are my findings:

1) No visual feedback after creating a new user or group (there should be a toast message)
2) Saving changes in the users control panel does not work (the edit bar does not show the save/cancel buttons, could be this is a side effect of me merging master branch):

Users and Groups

**Comment by nileshgulia1** [comment]
@tisto I fixed the above issues.

**Comment by tisto** [comment]
@nileshgulia1 thank you Nilesh!

File: CHANGELOG.md [hunk 2]
```diff
@@ -528,6 +528,15 @@
 - Drag and drop to reorder tiles @robgietema
 - Enhanced DraftJS AnchorLink Plugin @robgietema @sneridagh
 - Added the configuration required in Webpack config to load CSS modules in the project, required by DraftJS AnchorLink plugin @sneridagh
+* Refactor UserControlPanel add delete and add user actions @csenger
```
Comment (tisto) [inline CHANGELOG.md]: @nileshgulia1 This does not seem right. Can you re-add your changelog to the latest unreleased block on the changelog?

Comment (nileshgulia1) [reply CHANGELOG.md]: @tisto I have added the changelog!

Comment (tisto) [reply CHANGELOG.md]: @nileshgulia1 you still have to remove those duplicate changelog entries here.

Comment (nileshgulia1) [reply CHANGELOG.md]: Sorry. I removed the old entries now.

**Comment by tisto** [comment]
@nileshgulia1 when you merged master into your branch you introduced two new checks that are now failing on Travis:

1) eslint warnings: we fail the build on warnings now. check the output of "yarn lint" locally and fix the problems indicated
2) i18n warnings: we fail the build when the i18n is outdated, just run "yarn i18n" locally and check in the changes, this should fix that problem on Travis.

**Comment by nileshgulia1** [comment]
@tisto Should I commit the result of `yarn i18n`?

**Comment by tisto** [comment]
@nileshgulia1 yes, please.

**Comment by tisto** [comment]
@nileshgulia1 I updated the i18n. Let's see what the next build will return.

**Comment by nileshgulia1** [comment]
@tisto All Green!

File: locales/de/LC_MESSAGES/volto.po [hunk 3]
```diff
@@ -26,6 +26,14 @@ msgstr ""
 "Preferred-Encodings: utf-8 latin1\n"
 "X-Is-Fallback-For: de-at de-li de-lu de-ch de-de\n"
 
+#: components/manage/Controlpanels/UsersControlpanel
+msgid "ADD NEW GROUP"
```
Comment (tisto) [inline locales/de/LC_MESSAGES/volto.po]: @nileshgulia1 the message string should not be all upper-case. You can make the string appear all upper-case with CSS.

File: .gitignore [hunk 4]
```diff
@@ -15,7 +15,8 @@ build
 .DS_Store
 .idea
 lighthouse-report.html
-.vscode/
```
Comment (tisto) [inline .gitignore]: @nileshgulia1 you might want to re-add this line here for vscode users.

**Comment by tisto** [comment]
@nileshgulia1 thank you for your work on this! Merged. Would you mind creating another PR with the two small fixes I told you about!

**Comment by nileshgulia1** [comment]
@tisto okay! I'll do it asap.

**Pull request merged**